spec/lib/jobs/dell-wsman-bios-spec.js

```diff
@@ -0,0 +1,236 @@
+            });
+        });
+
+        it('Should fail to get respone from smi service', function(){
+            job.wsman = new WsmanTool();
+            job.inventories = ['bios', 'boot'];
+            job.dellConfigs = {
+                "wsmanCallbackUri": "http://172.31.128.1:9988/api/2.0/wsmanCallback/_IDENTIFIER_",
+                "services": {
+                    "inventory": {
+                        "serverCallback": "/api/1.0/server/inventory/callback"
+                    }
+                }
+            };
+            var response = {
+                "body": {
+                    "response": "Request response"
+                }
+            };
+            WsmanTool.prototype.clientRequest.resolves(response);
+            return job._handleAsyncRequest().then(function(){
+                expect(response.body.response.indexOf('Submitted')).to.equal(-1);
+            }).catch(function(error){
+                expect(error.name).to.equal("BadRequestError");
+                expect(error.message).to.equal("bios/boot request failed for node: undefined");
+            });
+        });
+    });
+
+    describe('handle cases of the function handleAsyncResponse', function(){
+        it('Should get error: data is invalid and scheduling retry', function(){
+            var result = {};
+            job.inventories = ['bios', 'boot'];
+            job.retrys = job.inventories.slice();
+
+            return job.handleAsyncResponse(result, "bios").then(function(){
+                expect(job.inventories.length).to.equal(3);
+                expect(job.inventories[2][0]).to.equal('bios');
+            });
+        });
+
+        it('Should get error: data is invalid.  No catalog created.', function(){
+            var result = {};
+            job.inventories = ['bios', 'boot'];
+            job.retrys = job.inventories.slice();
+
+            return job.handleAsyncResponse(result, "abc").then(function(){
+                expect(job.inventories.length).to.equal(2);
+            });
+        });
+
+        it('Should handlereponse: catalog not found and create', function(){
+            var result = {
+                "currentBootMode": "BIOS"
+            };
+            var catalog = {};
+            waterline.catalogs.findLatestCatalogOfSource.resolves(catalog);
+
+            return job.handleAsyncResponse(result, "bios").then(function(){
+                expect(result.currentBootMode).to.equal("BIOS");
+            });
+        });
+
+        it('Should handlereponse: catalog found and update', function(){
+            var result = {
+                "currentBootMode": "BIOS"
+            };
+            var catalog = {
+                "id": uuid.v4()
+            };
+            waterline.catalogs.findLatestCatalogOfSource.resolves(catalog);
+
+            return job.handleAsyncResponse(result, "bios").then(function(){
+                expect(result.currentBootMode).to.equal("BIOS");
```

Comment:
It doesn't make sense here since `result` is not returned from `handleAsyncResponse`.